Doc/library/asyncio-task.rst

```diff
@@ -624,32 +624,29 @@ Timeouts
     The context manager produced by :func:`asyncio.timeout` can be
     rescheduled to a different deadline and inspected.
 
-    .. class:: Timeout()
+    .. class:: Timeout(when)
 
        An :ref:`asynchronous context manager <async-context-managers>`
-       that limits time spent inside of it.
+       for cancelling overdue coroutines.
 
-        .. versionadded:: 3.11
+       The ``when`` parameter specifies at what time the context should
+       time out. It's value should be either a float relative to the
+       current :meth:`loop.time` or ``None``.
+
+       If ``when`` is ``None``, the timeout will never trigger.
+       If ``when < loop.time()``, the timeout will trigger on the next
+       iteration of the event loop.
```

Comment:
Slightly more succinct. We already state in the following two sentences what `when` does (and it should be pretty obvious from the parameter name anyway 😄); we don't need to say it in a vaguer way in the first sentence, I don't think.

Also: I liked the bullet points you used in the docstring! Let's use them again here:

```suggestion
       ``when`` should be either a float relative to the current
       :meth:`loop.time` or ``None``:

       - If ``when`` is ``None``, the timeout will never trigger.
       - If ``when < loop.time()``, the timeout will trigger on the next
         iteration of the event loop.
```